src/spirc.rs

```diff
@@ -170,6 +172,16 @@ impl<D: SpircDelegate> SpircManager<D> {
             protocol::spirc::MessageType::kMessageTypePause => {
                 self.delegate.pause();
             }
+            protocol::spirc::MessageType::kMessageTypeNext => {
+            	self.index = (self.index + 1) % self.tracks.len() as u32;
+                let track = self.tracks[self.index as usize];
```

Comment:
This is not 100% accurate I think.
When sending a Load command, the controlling device sends a context uri and small (~50 IIRC) list of tracks from that context.

For now librespot ignores the context uri, and only uses that list (this is on my TODO list to change but depends on lots of other stuff). Thus if the playlist is > 50 songs it will loop on the ones that were sent.